rotkehlchen/serialization/deserialize.py

```diff
@@ -450,6 +451,38 @@ def deserialize_optional(input_val: X | None, fn: Callable[[X], Y]) -> Y | None:
     return fn(input_val)
 
 
+def _get_transaction_receipt(
+        tx_hash: EVMTxHash,
+        raw_receipt_data: dict[str, Any] | None,
+        chain_id: ChainID,
+        timestamp: Timestamp,
+        evm_inquirer: 'EvmNodeInquirer',
+) -> dict[str, Any]:
+    """Get the transaction receipt for a tx during deserialization.
+    Handles a special case for Optimism where the official mainnet node should be tried first
+    no matter the default call order, since other nodes may have null L1 fee values.
+    """
+    if raw_receipt_data is not None:
+        return raw_receipt_data
+
+    call_order = None
+    if chain_id == ChainID.OPTIMISM and timestamp < OP_BEDROCK_UPGRADE:
+        mainnet_node = None
+        for node in (call_order := evm_inquirer.default_call_order()):
+            if node.node_info.endpoint.startswith('https://mainnet.optimism.io'):
+                mainnet_node = node
+                break
+
+        if mainnet_node is not None:
+            call_order.remove(mainnet_node)
+            call_order.insert(0, mainnet_node)
```

Comment:
can'y you get the index and just do a swap of indexeses?

```
call_order[0], call_order[i] = call_order[i], call_order[0]
```

moreover instead of the loop isn't it better doing an index_of like query https://docs.python.org/3/tutorial/datastructures.html#more-on-lists